node/node.go

```diff
@@ -208,3 +267,21 @@ func newTxValidator(pool mempool.Mempool, poolIDs *mempoolIDs, logger log.Logger
 		return checkTxResp.Code == abci.CodeTypeOK
 	}
 }
+
+func newHeaderValidator(logger log.Logger) p2p.GossipValidator {
+	return func(headerMsg *p2p.GossipMessage) bool {
+		logger.Debug("header received", "from", headerMsg.From, "bytes", len(headerMsg.Data))
+		var header types.Header
+		err := header.UnmarshalBinary(headerMsg.Data)
```

Comment:
Is there any way, we can only do the serialzation once? Currently this is done during validation and again in the `headerReadLoop`.

Reply:
I had the same concern. After discussing with Hlib I opened #126 where this is described.